### PR TITLE
Angular 15 compatibility

### PR DESCRIPTION
**Pull request by fanil-kashapov**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the new behavior?**
Update to angular 15

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by fh1ch — CHANGES_REQUESTED** [review]
@fanil-kashapov thanks a ton for this great MR, we appreciate this a lot 🙇 

A few general input/questions from my end, otherwise this one is good to go. Let me know what you think, back to you 🏓

File: angular.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
   "cli": {
-    "packageManager": "npm"
+    "packageManager": "npm",
+    "analytics": "bb52b033-97ca-4c6f-bbce-8db89400f162"
```
[the remark this answers is not in the document]
Comment (fanil-kashapov) [reply angular.json:5]: will disable,  it was automatically added by  'ng update' command

File: tsconfig.json [hunk 2]
```diff
@@ -10,16 +10,26 @@
     "module": "es2020",
```
[the remark this answers is not in the document]
Comment (fanil-kashapov) [reply tsconfig.json:10]: all updates to ES were provided by 'ng update' command, should I move it back to es2018?

File: package.json [hunk 3]
```diff
@@ -28,40 +28,43 @@
...
-    "@angular/router": "~13.1.0",
-    "eslint-plugin-import": "^2.25.3",
+    "@angular/animations": "^15.2.9",
+    "@angular/common": "^15.2.9",
+    "@angular/compiler": "^15.2.9",
+    "@angular/core": "^15.2.9",
+    "@angular/forms": "^15.2.9",
+    "@angular/platform-browser": "^15.2.9",
+    "@angular/platform-browser-dynamic": "^15.2.9",
+    "@angular/router": "^15.2.9",
```
[the remark this answers is not in the document]
Comment (fanil-kashapov) [reply package.json]: Done

**Review by fanil-kashapov — COMMENTED** [review]
Thx for review, done

**Comment by rampency** [comment]
can we merge this?

**Comment by supremeqwert** [comment]
Can this pr address the current issue with importing styles into css present in https://github.com/swimlane/ngx-datatable/issues/2162 ? Also what remains for this to be merged @fh1ch

**Comment by thatcort** [comment]
Hi, can this PR be merged?

**Review by fh1ch — APPROVED** [review]
Huge thanks for the rework and sorry for the delay here 🙇 

All findings addressed and no new questions from my end, so let's get this released.

**Pull request merged**